### PR TITLE
[FEAT] Collapse code sections

### PR DESCRIPTION
**Pull request by SimonFuet**

### CODE REVIEW

File: src/tools.ts [hunk 1]
```diff
@@ -220,3 +220,50 @@ export function hideCodeWithoutDecoration (editor: monaco.editor.IStandaloneCode
...
+    editor.setSelections(ranges.map(r => ({
+      selectionStartLineNumber: r.startLineNumber,
+      selectionStartColumn: r.startColumn,
+      positionLineNumber: r.endLineNumber,
+      positionColumn: r.endColumn
+    })))
+    await editor.getAction('editor.createFoldingRangeFromSelection').run()
+    if (selections != null) {
+      editor.setSelections(selections)
+    }
```
Comment (samuel-olivier) [inline src/tools.ts:267]: Isn't there any other way than faking selection temporarily ? Does it blink visually ?

File: src/tools.ts [hunk 2]
```diff
@@ -220,3 +220,50 @@ export function hideCodeWithoutDecoration (editor: monaco.editor.IStandaloneCode
...
+    let currentPosition = editorModel.getFullModelRange().getStartPosition()
+    let match: monaco.editor.FindMatch | null
+    while ((match = editorModel.findNextMatch(startToken,
+      /* searchStart */currentPosition,
+      /* isRegex */isRegex,
+      /* matchCase */true,
+      /* wordSeparators */null,
+      /* captureMatches */false
+    )) != null) {
+      if (match.range.getStartPosition().isBefore(currentPosition)) {
```
Comment (samuel-olivier) [inline src/tools.ts:240]: how can the `match` be before the `currentPosition` as we set `searchStart` to `currentPosition` ?

File: src/tools.ts [hunk 3]
```diff
@@ -220,3 +220,50 @@ export function hideCodeWithoutDecoration (editor: monaco.editor.IStandaloneCode
...
+        break
+      }
+      const matchEnd = editorModel.findNextMatch(endToken,
+        /* searchStart */match.range.getEndPosition(),
+        /* isRegex */isRegex,
+        /* matchCase */true,
+        /* wordSeparators */null,
+        /* captureMatches */false
+      )
+      if (matchEnd != null && matchEnd.range.getStartPosition().isBefore(match.range.getStartPosition())) {
```
[the remark this answers is not in the document]
Comment (CGNonofr) [reply src/tools.ts:250]: Thats the trick, `findNextMatch` loops to the bottom if there is no match after `currentPosition`

File: src/tools.ts (hunk 2 above)
Comment (CGNonofr) [reply src/tools.ts:240]: see comment 1 (:

**Review by samuel-olivier — APPROVED** [review]

**Pull request merged**